src/tn_sdk/core/tn_api.py

```diff
@@ -1,9 +1,63 @@
+import json
+import zlib
+from base64 import b64encode
+
+from tn_sdk.exceptions.exceptions import InvalidDataException
+from tn_sdk.utils.constants import (
+    SANDBOX_API_URL,
+    PRODUCTION_API_URL,
+    DEFAULT_COMPRESSION_LEVEL,
+)
+
+
 class TnApi:
+    """
+    The entrypoint to the Trip Ninja SDK. Exposes useful functionality of the Trip Ninja API to the end user.
+    """
+
+    _TN_BASE_PROD_URL: str = PRODUCTION_API_URL
+    _TN_BASE_SANDBOX_URL: str = SANDBOX_API_URL
+    _GZIP_DEFAULT_COMPRESSION_LEVEL: int = DEFAULT_COMPRESSION_LEVEL
+
+    def __init__(self, access_token: str, refresh_token: str, is_sandbox: bool = False):
+        """
+        Initializes and instantiates the SDK client.
+
+        :param access_token: The token provided by the Admin Panel that is used to gain access to the API.
+        :param refresh_token: The token provided by the Admin Panel that is used to refresh a user's access token.
+        :param is_sandbox: Whether the client is connecting to the sandbox or the production environment.
+        """
+        self._access_token: str = access_token
+        self._refresh_token: str = refresh_token
+        self.base_url: str = (
+            self._TN_BASE_SANDBOX_URL if is_sandbox else self._TN_BASE_PROD_URL
+        )
 
     def prepare_data_for_generate_solutions(self, json_data: str) -> bytes:
         """
         This function prepares the data for generating solutions by compressing the data
+
         :param json_data: JSON encoded data
         :return: compressed bytes
         """
-        pass
+
+        if not isinstance(json_data, str):
+            raise InvalidDataException("Input must be a JSON-encoded string")
+
+        try:
+            json.loads(json_data)
+        except json.JSONDecodeError as e:
+            raise InvalidDataException(
+                f"Input is not valid JSON: {str(e)}"
+            ) from e
```

Comment:
I like the idea behind this extra check, but IMO the cost of performing a `json.loads` only to check for proper format is too high to justify it. I think given the variable name and docstring that should hopefully be enough. 
A potential different way to tackle this would be to add some extra detail to the error msg we throw, like "did you make sure to json.loads your data before hand"
or we maybe could even do the json.loads ourself if we are worried about users having trouble on this step.

lemme know what you think, I have a open mind here if you think this check is worth it.